api/database_operations.py

```diff
@@ -0,0 +1,63 @@
+import pyodbc
+from configparser import ConfigParser
+from datetime import datetime
+
+config_object = ConfigParser()
+config_object.read("config.ini")
+
+server = config_object['SETUP']['server']
+database = config_object['SETUP']['database']
+username = config_object['SETUP']['username']
+password = config_object['SETUP']['password']
+driver = config_object['SETUP']['driver']
+port = config_object['SETUP']['port']
+
+
+class DBOperations():
+
+    def connection(self):
+        connection = pyodbc.connect('DRIVER=' + driver + ';SERVER=' + server + ';PORT=' + port + ';DATABASE=' + database + ';UID=' + username + ';PWD=' + password)
+        return connection
+
+    def get_all_scores(self):
+        scores = []
+        query = '''
+                SELECT * FROM telemed_users;
+                '''
+
+        conn = self.connection()
+        cursor = conn.cursor()
+        cursor.execute(query)
+        columns = [desc[0] for desc in cursor.description]
+
+        for row in cursor.fetchall():
+            row_dict = dict(zip(columns, row))
+            scores.append(row_dict)
+        conn.close()
+        return scores
+
+
+    def get_best_scores(self, num_of_scores):
+        scores = []
+        query = '''
+                SELECT * FROM telemed_users ORDER BY Score;
+                '''
+
+        conn = self.connection()
+        cursor = conn.cursor()
+        cursor.execute(query)
+        columns = [desc[0] for desc in cursor.description]
+
+        for row in cursor.fetchall():
+            row_dict = dict(zip(columns, row))
+            scores.append(row_dict)
+        conn.close()
+        return scores[:num_of_scores]
```

Comment:
Tutaj wydaje mi się, że trzeba obsłużyć przypadek kiedy lista scores jest krótsza niż num_of_scores. Ja u siebie w branchu już to edytowałem na:
        if len(scores) < num_of_scores:
            return scores
        else:
            return scores[:num_of_scores]
Chyba ok, nie?

Reply:
Wydaje się spoko

Reply:
Git